utils/techutils/techutils.go

```diff
@@ -507,3 +532,68 @@ func GetAllTechnologiesList() (technologies []Technology) {
 	}
 	return
 }
+
+// SplitComponentId splits a Xray component ID to the component name, version and package type.
+// In case componentId doesn't contain a version, the returned version will be an empty string.
+// In case componentId's format is invalid, it will be returned as the component name
+// and empty strings will be returned instead of the version and the package type.
+// Examples:
+//  1. componentId: "gav://antparent:ant:1.6.5"
+//     Returned values:
+//     Component name: "antparent:ant"
+//     Component version: "1.6.5"
+//     Package type: "Maven"
+//  2. componentId: "generic://sha256:244fd47e07d1004f0aed9c156aa09083c82bf8944eceb67c946ff7430510a77b/foo.jar"
+//     Returned values:
+//     Component name: "foo.jar"
+//     Component version: ""
+//     Package type: "Generic"
+//  3. componentId: "invalid-comp-id"
+//     Returned values:
+//     Component name: "invalid-comp-id"
+//     Component version: ""
+//     Package type: ""
+func SplitComponentId(componentId string) (string, string, string) {
+	compIdParts := strings.Split(componentId, "://")
+	// Invalid component ID
+	if len(compIdParts) != 2 {
+		return componentId, "", ""
+	}
+
+	packageType := compIdParts[0]
+	packageId := compIdParts[1]
+
+	// Generic identifier structure: generic://sha256:<Checksum>/name
+	if packageType == "generic" {
+		lastSlashIndex := strings.LastIndex(packageId, "/")
+		return packageId[lastSlashIndex+1:], "", packageTypes[packageType]
+	}
+
+	var compName, compVersion string
+	switch packageType {
+	case "rpm":
+		// RPM identifier structure: rpm://os-version:package:epoch-version:version
+		// os-version is optional.
+		splitCompId := strings.Split(packageId, ":")
+		if len(splitCompId) >= 3 {
```

Comment:
are you certain about this condition? if len(splitCompId) > 3 so the compName (which is the package name for my understanding) might not be in this index maybe?